ln-gateway/src/lib.rs

```diff
@@ -282,6 +279,12 @@ impl LnGateway {
     }
 
     pub async fn run(&mut self) -> Result<()> {
+        // Regster gateway with federation
+        self.federation_client
+            .register_with_federation(self.federation_client.config().into())
+            .await
+            .expect("Failed to register with federation");
```

Comment:
We probably shouldn't `expect` here. In principle, if the gateway can't connect with the federation it can't really run. So we might want to exit, but probably shouldn't panic. But we're `expect`ing currently so this PR seems like definite improvement to me.